addons/mail/wizard/mail_activity_schedule_wizard.py

```diff
@@ -0,0 +1,315 @@
+        for wizard in self:
+            if wizard.plan_id and not self.env['mail.activity.plan'].search(
+                    expression.AND([wizard._get_plan_domain({
+                        'company_id': wizard.company_id.id,
+                        'res_model_id': wizard.res_model_id.id,
+                        'records': wizard._get_records(),
+                    }), [('id', '=', wizard.plan_id.id)]])):
+                raise ValidationError(_('Incompatible plan'))
+
+    def _get_errors_plan(self):
+        self.ensure_one()
+        errors = set()
+        for record in self._get_records():
+            for activity_template in self.plan_id.template_ids:
+                error = activity_template._get_responsible_id(self.on_demand_user_id, record)['error']
+                if error:
+                    errors.add(error)
+        return errors
+
+    def _get_errors(self):
+        self.ensure_one()
+        if self.plan_id:
+            return self._get_errors_plan()
+        errors = set()
+        if not self.user_id:
+            errors.add(_('Responsible is required'))
+        if not self.activity_type_id:
+            errors.add(_('Activity type is required'))
+        return errors
+
+    @api.depends('company_id', 'res_model_id', 'res_ids')
+    def _compute_has_plan(self):
+        ActivityPlan = self.env['mail.activity.plan']
+        company_ids = {r.company_id.id for r in self}
+        res_model_ids = {r.res_model_id.id for r in self}
+        domain = expression.AND([
+            expression.OR([[('company_id', 'in', tuple(company_ids))], [('company_id', '=', False)]]),
+            expression.OR([[('res_model_ids', '=', False)],
+                           *[[('res_model_ids', '=', res_model_id)] for res_model_id in res_model_ids]])
+        ])
+        plan_ids_by_company_res_model = {
+            (company_id.id or False, res_model_id or False): cnt
+            for company_id, res_model_id, cnt in ActivityPlan._read_group(
+                domain, groupby=["company_id", "res_model_ids"], aggregates=["__count"])
+        }
+        for record in self:
+            record.has_plan = any(
+                plan_ids_by_company_res_model.get(company_res_model_id)
+                for company_res_model_id in ((False, False),
+                                             (False, record.res_model_id),
+                                             (record.company_id.id, False),
+                                             (record.company_id.id, record.res_model_id))
+            )
+
+    @api.depends('plan_id', 'res_model_id', 'res_ids',
+                 'on_demand_user_id',  # plan specific
+                 'activity_type_id', 'user_id')  # activity specific
+    def _compute_error(self):
+        for wizard in self:
+            errors = wizard._get_errors()
+
+            if errors:
+                if self.plan_id:
+                    error_display = _('The plan "%s" cannot be launched: <br><ul>', wizard.plan_id.name)
```

Comment:
Would escape plan name or use Markup, depending on which is easier :) (did not look in details at 'errors' depending on it could be one or another solution)